service/history/shard/context_impl.go

```diff
@@ -1534,7 +1534,12 @@ func (s *ContextImpl) ioSemaphoreAcquire(
 		}
 	}()
 
-	return s.ioSemaphore.Acquire(ctx, 1)
+	priority := locks.PriorityHigh
+	callerInfo := headers.GetCallerInfo(ctx)
+	if callerInfo.CallerType == headers.CallerTypePreemptable {
```

Comment:
👍 
I was original thinking we need to use low priority for workflow timeout timer as well, but now that I think about it with my execution timeout timer change we should not run into hot workflow issue due to the timeout timer.

Reply:
Also, we can easily add more priority levels to this semaphore. If we need that in the future, we can use the same priority levels we use in the persistence rate limiter.